### ISSUE TITLE
Implement IntoResponse for Result

### ISSUE THREAD

**Issue by nyarly** [gotham-rs/gotham#72]
If this were in Gotham:
```rust
impl IntoResponse for Result<Response> {
  fn into_response(self, state: &State) -> Response {
    match self {
      Ok(res) => res.into_response(state),
      Err(e) => e.into_response(state),
    }
  }
}
```

It would simplify some middleware code. For instance I have this in one of my middleware:

```rust
          match result {
            Ok(r) => (state, r).into_handler_future(),
            Err(e) => {
              error!("Require Authn: {}", e);
              let res = Response::new()
                .with_status(StatusCode::InternalServerError)
                .with_body(format!("Require Authn: {}", e))
                .with_header(ContentType::plaintext());
              (state, res).into_handler_future()
            }
          }
```

If Result<Response,Error> were IntoResponse, the above could be reduced to 

```rust
(state,result).into_handler_future()
```

I'm volunteering to address this is with a PR, but wanted to present the case and record the intention.

**Comment by bradleybeddoes** [comment]
I'm 👍 for seeing what this looks like in a PR @nyarly

**Comment by bradleybeddoes** [comment]
Closed by #99
